modules/lists/cow_list.py

```diff
@@ -0,0 +1,18 @@
+
+
+class CowList:
+    def __init__(self, input_list):
+        self.backing_list = input_list[:]
+        
+    def __getitem__(self, i):
+        return self.backing_list[i]   
+
+    def append(self, item):
+        modified_list = self.backing_list[:]
+        modified_list.append(item)
+        return CowList(modified_list)
+
+    def set_value(self, i, value):
```

Comment:
What happens if you do x[0] = 4 for your cow list? It looks like you need to override that method and raise an exception!